cli/plasmo/src/features/extension-devtools/project-creator.ts

```diff
@@ -0,0 +1,151 @@
+import spawnAsync from "@expo/spawn-async"
+import { copy, readJson, writeJson } from "fs-extra"
+import { writeFile } from "fs/promises"
+import { userInfo } from "os"
+import { resolve } from "path"
+import { temporaryDirectory } from "tempy"
+
+import { hasFlag, iLog, vLog } from "@plasmo/utils"
+
+import { generateGitIgnore } from "~features/extension-devtools/git-ignore"
+import {
+  PackageJSON,
+  generatePackage,
+  resolveWorkspaceToLatestSemver
+} from "~features/extension-devtools/package-file"
+import {
+  TemplatePath,
+  getTemplatePath
+} from "~features/extension-devtools/template-path"
+import type { PackageManagerInfo } from "~features/helpers/package-manager"
+
+import type { CommonPath } from "./common-path"
+
+const regex = /--with-[\w|-]+/
```

Comment:
This will match

`--with--` is this what you want? 

If not, `/--with-\w+/`

Reply:
`--with-(?:\w+-?)+([^-]$)`

i think this should do it for you lemme know if you want me to tweak it

Reply:
I tweaked it to this: 
`/(?:^--with-)(?:\w+-?)+(?:[^-]$)/`

Which will make sure the --with must be at the start. This doesn't match `with-s` for some reason lol, but whatever we shall ship it